File: src/main/java/ru/surpavel/bugtrackingsystem/service/IService.java
package ru.surpavel.bugtrackingsystem.service;

import java.util.List;
import java.util.Optional;

public interface IService<T> {

    T save(T entity);

    Optional<T> findById(Long id);

    boolean existsById(Long id);

    List<T> findAll();

    void delete(T entity);
}

File: src/main/js/angularclient/src/app/service/iservice.ts
import { Observable } from "rxjs"

export interface Iservice<T> {
    
    save(entity: T): Observable<T>;

    findById(id: number): Observable<T>;

    findAll(): Observable<T[]>;

    update(id: number, entity: T): Observable<T>;

    delete(id: number): Observable<T>;

}
